pkg/services/publicdashboards/api/api.go

```diff
@@ -73,10 +73,15 @@ func (api *Api) RegisterAPIEndpoints() {
 		auth(middleware.ReqSignedIn, accesscontrol.EvalPermission(dashboards.ActionDashboardsRead, uidScope)),
 		routing.Wrap(api.GetPublicDashboard))
 
-	// Create/Update Public Dashboard
+	// Create Public Dashboard
 	api.RouteRegister.Post("/api/dashboards/uid/:dashboardUid/public-dashboards",
 		auth(middleware.ReqOrgAdmin, accesscontrol.EvalPermission(dashboards.ActionDashboardsPublicWrite, uidScope)),
-		routing.Wrap(api.SavePublicDashboard))
+		routing.Wrap(api.CreatePublicDashboard))
+
+	// Update Public Dashboard
+	api.RouteRegister.Put("/api/dashboards/uid/:dashboardUid/public-dashboards",
```

Comment:
We should update a public dashboard by a public dashboard ID ni the URL like delete.

Reply:
Agreed.